modules/rostests/apitests/shell32/ShellExecuteEx.cpp

```diff
@@ -81,8 +155,67 @@ TestShellExecuteEx(const WCHAR* Name, BOOL ExpectedResult)
     }
 }
 
+static
+VOID
+TestShellExecuteEx1(const WCHAR* Name, const WCHAR* Params)
+{
+    SHELLEXECUTEINFOW ShellExecInfo;
+    BOOL Result;
+    CHAR OSVendor[100] = {0};
+    INT iResult;
+    UINT_PTR retval = SE_ERR_NOASSOC;
+
+    ZeroMemory(&ShellExecInfo, sizeof(ShellExecInfo));
+    ShellExecInfo.cbSize = sizeof(ShellExecInfo);
+    ShellExecInfo.fMask = SEE_MASK_NOCLOSEPROCESS | SEE_MASK_FLAG_NO_UI;
+    ShellExecInfo.hwnd = NULL;
+    ShellExecInfo.nShow = SW_SHOWNORMAL;
+    ShellExecInfo.lpFile = Name;
+    ShellExecInfo.lpParameters = Params;
+    ShellExecInfo.lpDirectory = NULL;
+    ShellExecInfo.lpVerb = L"open";
+
+    iResult = GetOSVendor(OSVendor, sizeof(OSVendor));
+    printf("OSVendor is '%s'.\n", OSVendor);
+
+    if(strcmp(OSVendor, "Microsoft") == 0)
+        skip("Test does not work on MS Windows.\n");
+
+    if(iResult == FALSE)
```

Comment:
This should happen earlier.

Reply:
@SergeGautherie, I disagree. This statement and the previous one could occur is this order or reversed as you suggest and still result in the same reporting and outcome.

Reply:
I usually do not check invalid data, but if that's what you want to do...